src/silverlabnwb/signature.py

```diff
@@ -250,6 +306,13 @@ def ignored_path(self, path):
                 return True
         return False
 
+    def should_cast_path(self, path, encountered_type):
+        """Should we cast this entity path, and if so, to what type?"""
+        for key in self._cast_paths:
```

Comment:
Slightly better would be `for key, types in self._cast_paths.items():` then you can use `types['expected']` etc below.